### PR TITLE
Dynamically update `checked` prop for Switch component

### LINKED ISSUE #860: Switch does not handle attribute updates
### Description

The Switch component can be used perfectly, when it is clicked directly. The lever will toggle as expected. But in case that the switch is toggled via a different component (e.g. two switches share the same state) the "not-clicked" lever does not updated - although the checked attribute changes correctly in the DOM. 

### Steps to Reproduce

1. click on one of the Switches - only the clicked lever will update (visually)
```
constructor(props) {
  super(props);
  this.state = {
    cbk: false
  }
}
```

```	
render(){
  return(
    <div>
      <Switch
        checked={this.state.cbk}
        offLabel={"off"}
        onLabel={"on"}
        onChange={() => {
          this.setState({cbk: !this.state.cbk});
        }}
      />
      <Switch
        checked={this.state.cbk}
        offLabel={"off"}
        onLabel={"on"}
        onChange={() => {
          this.setState({cbk: !this.state.cbk});
        }}
      />
    </div>
  )
}
```

**Expected behavior:** 

As both switches refer to the same state both levers should be updated

**Actual behavior:** 

1) only the clicked lever updates automatically
2) both input fields will get the "checked" attribute

### Versions

**react-materialize:** 3.2.2
**materialize-css:** 1.0.0
**react:** 16.8.6

### PR DESCRIPTION
**Pull request by CloudPower97**
# Description

This __P.R__ fixes #860 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested with `Formik`

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by alextrastero** [comment]
Also, let's pass rest of props instead of classname ?

**Comment by CloudPower97** [comment]
> Shouldn't we ALSO have `defaultChecked`? https://reactjs.org/docs/uncontrolled-components.html

Not sure about that... I mean, we have to choose between `controlled` or `uncontrolled` component.
I do not think we can implement both things.
But as I said before, I'm not completely sure about that.

Anyway, I've spreaded all `prop`s now.
I've also used `defaultProps` for `id` and removed `Component` import from `React` which is not used at all.
Have a look at this whenever you can :smile:

File: src/Switch.js [hunk 1]
```diff
@@ -52,4 +59,8 @@ Switch.propTypes = {
   checked: PropTypes.bool
 };
 
+Switch.defaultProps = {
+  id: idgen()
```
Comment (alextrastero) [inline src/Switch.js]: 👍

File: src/Switch.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+      <label htmlFor={id}>
         {offLabel}
         <input
-          id={computedId}
+          id={id}
           disabled={disabled}
           onChange={onChange}
           type="checkbox"
-          defaultChecked={checked}
+          checked={checked}
```
Comment (alextrastero) [inline src/Switch.js]: since we have spread, do we need `id, disabled, onChange, checked`

Comment (CloudPower97) [reply src/Switch.js]: Yes, we have to keep these `prop`s as is

**Comment by CloudPower97** [comment]
> Regarding `defaultChecked` controlled vs uncontrolled I agree, but for `TextInput` we pass both, is this good?

According to [`React` docs ](https://reactjs.org/docs/uncontrolled-components.html)

> With an uncontrolled component, you often want React to specify the initial value, but leave subsequent updates uncontrolled.

So it seems that the `defaultChecked` `prop` is only used during initial render.

I'm still not sure if we can use both `defaultChecked` and `checked`.

To be honest, I'm inclined to the `controlled` component approach, so I think we should simply drop `defaultChecked`, also because one can achieve the same by setting `checked` to `true` or `false`, depending on your needs, mimicking `defaultChecked` or `defaultValue` in the case of `TextInput`.

What do you think we should do? :smiley:

**Comment by CloudPower97** [comment]
@alextrastero ping :smile:

File: src/Switch.js (hunk 1 above)
Comment (alextrastero) [inline src/Switch.js]: `idgen` returns a number, let's convert it to a string

**Comment by alextrastero** [comment]
Yes agreed, I would prefer a controlled component, but then we'll need a default prop of `onChange: () => {}` to remove the warning.

**Comment by alextrastero** [comment]
Maybe a follow up issue ticket to investigate if this can also be done in `<TextInput />`

**Comment by alextrastero** [comment]
👍

**Pull request merged**